llama_stack/providers/inline/agents/meta_reference/agent_instance.py

```diff
@@ -543,7 +541,12 @@ async def _run(
                 # if the session has a memory bank id, let the memory tool use it
                 if session_info.vector_db_id:
                     vector_db_ids.append(session_info.vector_db_id)
+                elif vector_db_ids:
+                    # In case of multiple vector DBs, we use the first one
```

Comment:
This seems somewhat arbitrary; I think we should at least log that information. What do you think?

Reply:
Agree with logging of course.
About the "arbitrary" part: what else could we do in this case? Some ideas that come to mind:
- add an explicit config arg to identify the ingestion vector_db?
- extend the concept of session vector_db to store a list of ids?
- stop the execution in case more dbs are given? (when documents are also provided)
- other?

Reply:
Thanks for the suggestions!

I don’t think the code should be making decisions on behalf of the user, so having a config arg to specify which DB to use in this scenario makes sense to me.

Reply:
I think an important use case for multiple vector DBs is federating across those vector DBs.  That's challenging to do well, and the approach to doing it is different when each vector DB has the same content and when they have different content.  When they have the same content, you generally want to specify some sort of unique ID on each chunk and/or document that you can use to recognize when the same result came from two different sources so you can boost that result.  All of that is out of scope for this PR of course, but it would be good to design the configuration for which DB to use in a way that reflects that in the future we might want the users to be able to select all and/or a subset and then provide additional configuration details about how to federate across them.

Reply:
@jwm4 👍 for `that is out of scope for this PR`

Reply:
@leseb I've added a new field `insert_vector_db_id` to configure the ingestion DB but I'm wondering how we can document this change. I see that the other argument `vector_db_ids` has not been well documented either, but just mentioned in code snippets.